File: vue/src/domain/cost-model.ts
export interface CostInput {
  area_mode: 'flat' | 'faces' | 'layout' | 'total'
  width_m: number; height_m: number; area_m2: number; thickness_mm: number
  pieces: number; hole_count: number; process_length_m: number
  faces: Array<{ width_m: number; height_m: number; count: number }>
  processes: Record<string, boolean>
}
export interface CostModel {
  version: number; material: string; basis: 'area' | 'piece' | 'set' | 'manual'
  unit_price: number; multiplier: number; minimum: number
  small_area_threshold: number; small_area_surcharge: number
  thickness_prices: Array<{ thickness_mm: number; unit_price: number }>
  processes: Array<{code: string; unit: string; unit_price: number; multiplier: number}>
}
export const processNames: Record<string,string> = {slot:'开槽',punch:'打孔',laminate:'覆膜',double_sided:'双面'}
export function emptyCostInput(): CostInput {
  return {area_mode:'total',width_m:0,height_m:0,area_m2:0,thickness_mm:0,pieces:1,hole_count:0,process_length_m:0,faces:[],processes:{}}
}
export function emptyCostModel(): CostModel {
  return {version:1,material:'',basis:'area',unit_price:0,multiplier:1,minimum:0,small_area_threshold:0,small_area_surcharge:0,thickness_prices:[],processes:[]}
}

File: domain/cost_model.go
package domain

// CostModel is a bounded configuration, never executable expression text.
// Unit costs always describe ONE SKU sales unit (one piece or one complete set).
type CostModel struct {
	Version            int                  `json:"version"`
	Material           string               `json:"material"`
	Basis              string               `json:"basis"` // area, piece, set, manual
	UnitPrice          float64              `json:"unit_price"`
	Multiplier         float64              `json:"multiplier"`
	Minimum            float64              `json:"minimum"`
	SmallAreaThreshold float64              `json:"small_area_threshold"`
	SmallAreaSurcharge float64              `json:"small_area_surcharge"`
	ThicknessPrices    []CostThicknessPrice `json:"thickness_prices,omitempty"`
	Processes          []CostProcessPrice   `json:"processes,omitempty"`
}
type CostThicknessPrice struct {
	ThicknessMM float64 `json:"thickness_mm"`
	UnitPrice   float64 `json:"unit_price"`
}
type CostProcessPrice struct {
	Code       string  `json:"code"`
	Unit       string  `json:"unit"` // sku, piece, hole, metre, area
	UnitPrice  float64 `json:"unit_price"`
	Multiplier float64 `json:"multiplier"`
}
type CostFace struct {
	WidthM  float64 `json:"width_m"`
	HeightM float64 `json:"height_m"`
	Count   int     `json:"count"`
}
type CostInput struct {
	AreaMode       string          `json:"area_mode"` // flat, faces, layout, total
	WidthM         float64         `json:"width_m"`
	HeightM        float64         `json:"height_m"`
	DepthM         float64         `json:"depth_m,omitempty"` // reference dimension; never guessed as layout
	AreaM2         float64         `json:"area_m2"`
	ThicknessMM    float64         `json:"thickness_mm"`
	Pieces         int             `json:"pieces"` // pieces within ONE SKU, not order quantity
	Faces          []CostFace      `json:"faces,omitempty"`
	Processes      map[string]bool `json:"processes,omitempty"`
	HoleCount      int             `json:"hole_count"`
	ProcessLengthM float64         `json:"process_length_m"`
}
type CostLine struct {
	Name       string  `json:"name"`
	Quantity   float64 `json:"quantity"`
	Unit       string  `json:"unit"`
	UnitPrice  float64 `json:"unit_price"`
	Multiplier float64 `json:"multiplier"`
	Amount     float64 `json:"amount"`
}
type CostCalculation struct {
	Status           string     `json:"status"` // calculated, missing_input, manual_quote, rule_conflict
	Input            CostInput  `json:"input"`
	AreaM2           float64    `json:"area_m2"`
	BillableQuantity float64    `json:"billable_quantity"`
	Lines            []CostLine `json:"lines"`
	Missing          []string   `json:"missing"`
}
